crates/sui-core/src/authority/authority_store.rs

```diff
@@ -786,7 +786,7 @@ impl AuthorityStore {
     #[instrument(level = "trace", skip_all)]
     async fn acquire_read_locks_for_indirect_objects(
         &self,
-        written: &WrittenObjects,
+        written: &[Object],
```

Comment:
Seems better to be `impl Iterator<Item = &Object>`
Then in the caller you could just use `iter` + `map` + `flatten`

Reply:
I tried to do this initially but it created intractable compile errors due to a weird interaction between the iterator lifetimes and the async blocks that call this fn